runtime/jvmti/jvmtiThread.c

```diff
@@ -1109,21 +1117,41 @@ resumeThread(J9VMThread *currentThread, jthread thread)
 		if (NULL != targetThread)
 #endif /* JAVA_SPEC_VERSION >= 19 */
 		{
+#if JAVA_SPEC_VERSION >= 19
+			if (0 == J9OBJECT_U32_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset)) {
+				rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+			} else {
+				J9OBJECT_U32_STORE(currentThread, threadObject, vm->isSuspendedInternalOffset, 0);
+			}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
+#if JAVA_SPEC_VERSION >= 19
+			if ((0 == J9OBJECT_U32_LOAD(currentThread, targetThread->threadObject, vm->isSuspendedInternalOffset))
+			&& (0 == J9OBJECT_U32_LOAD(currentThread, targetThread->carrierThreadObject, vm->isSuspendedInternalOffset)))
+			{
+				if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)) {
+					clearHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
+					Trc_JVMTI_threadResumed(targetThread);
+				} else {
+					rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+				}
+			}
+#else /* JAVA_SPEC_VERSION >= 19 */
 			if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)) {
 				clearHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
 				Trc_JVMTI_threadResumed(targetThread);
 			} else {
 				rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
 			}
+#endif /* JAVA_SPEC_VERSION >= 19 */
```

Comment:
This is not needed after the above ifdef modification.

Reply:
The unwanted ifdef and code parts are removed.